lte/gateway/c/core/oai/tasks/nas/esm/sap/esm_recv.c

```diff
@@ -523,13 +523,14 @@ esm_cause_t esm_recv_pdn_disconnect_request(
     OAILOG_FUNC_RETURN(LOG_NAS_ESM, ESM_CAUSE_INVALID_EPS_BEARER_IDENTITY);
   }
 
+  bool no_delete_tunnel = (ue_mm_context_p->nb_active_pdn_contexts > 1);
```

Comment:
this is always true due to line 470. But, I think we should set this always to false as we should not delete the gtpv2c tunnel until all sessions got deleted.